### PR TITLE
Small refactoring in coqPackages

### PR DESCRIPTION
**Pull request by vbgl**
###### Motivation for this change

Make it easier to override Coq packages.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/top-level/coq-packages.nix [hunk 1]
```diff
@@ -54,7 +54,8 @@ in rec {
 
   mkCoqPackages = coq:
     let self = mkCoqPackages' self coq; in
-    filterCoqPackages coq self;
+    if coq.doNotFilter or false then self
```
Comment (roberth) [inline pkgs/top-level/coq-packages.nix]:
```suggestion
    if coq.doNotFilter then self
```

Comment (roberth) [reply pkgs/top-level/coq-packages.nix]: Right! It's the `a.b or c` syntax. I can't believe I fell for that.

Do we really need the negated name? Negated names require unnecessary extra thinking. What about "setting `doFilter = false`" which has a default of true?

Comment (vbgl) [reply pkgs/top-level/coq-packages.nix]: Maybe Théo has an opinion. Ping @Zimmi48 .

Comment (Zimmi48) [reply pkgs/top-level/coq-packages.nix]: Ah I did not know about the syntax, will use ;-)
Regarding `doFilter = false` vs `doNotFilter = true`, it seems that the implicit convention in nixpkgs (https://nixos.org/nixpkgs/manual/#sec-stdenv-phases) is that when something is off by default, it is enabled with `doStuff = true` but when something is on by default, it is disabled `dontStuff = true`. Thus, it suggests to use `dontFilter` here.

Comment (vbgl) [reply pkgs/top-level/coq-packages.nix]: Let’s go with `dontFilter`.

**Pull request merged**